bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/XmlSequence.java

```diff
@@ -637,4 +640,27 @@ private Type getSequenceType(Type tempExprType) {
                 return PredefinedTypes.TYPE_TEXT;
         }
     }
+
+    private void initializeIteratorNextReturnType() {
+        Type childrenType;
+        LinkedHashSet<Type> types = new LinkedHashSet<>();
+        for (int i = 0; i < children.size(); i++) {
+            types.add(children.get(i).getType());
+        }
+        if (types.size() == 1) {
```

Comment:
Can we have empty types?

Reply:
Here can we check types size first and childrenType = children.get(0).getType() instead of using iterator?

Reply:
Suggestions addressed with https://github.com/ballerina-platform/ballerina-lang/pull/32210/commits/8ccf9f67387d4e4590c45df9f5008d3c5b9aed24